### PR TITLE
Create GH issue if there are errors while publishing on a release pipeline

### PR DESCRIPTION
**Pull request by jcagme**
Fixes: https://github.com/dotnet/arcade/issues/2303

### CODE REVIEW

File: src/Microsoft.DotNet.Build.Tasks.Feed/src/PushArtifactsInManifestToFeed.cs [hunk 1]
```diff
@@ -219,7 +257,94 @@ public async Task<bool> ExecuteAsync()
...
+                await CreateGitHubIssue(repoUrl, commit, buildId);
+            }
+
             return !Log.HasLoggedErrors;
         }
+
+        /// <summary>
+        /// Creates a new GitHub issue.
+        /// </summary>
+        /// <param name="repoUrl">GitHub source repo IRL.</param>
```
Comment (riarenas) [inline src/Microsoft.DotNet.Build.Tasks.Feed/src/PushArtifactsInManifestToFeed.cs]:
```suggestion
        /// <param name="repoUrl">GitHub source repo URL.</param>
```

Although I love that this implies that you can meet a repo IRL

File: src/Microsoft.DotNet.Build.Tasks.Feed/src/PushArtifactsInManifestToFeed.cs [hunk 2]
```diff
@@ -59,41 +98,35 @@ public class PushArtifactsInManifestToFeed : MSBuild.Task
...
-        /// Full path to the folder containing blob assets.
+        /// Title of the release which triggered this task.
         /// </summary>
-        [Required]
-        public string BlobAssetsBasePath { get; set; }
+        public string CurrentReleaseDescription { get; set; }
 
         /// <summary>
-        /// Full path to the folder containing package assets.
+        /// The URL of the build which triggered the release which triggered this task.
```
Comment (riarenas) [inline src/Microsoft.DotNet.Build.Tasks.Feed/src/PushArtifactsInManifestToFeed.cs]: [nit] This reads weird, but I couldn't come up with a better suggestion

Comment (jcagme) [reply src/Microsoft.DotNet.Build.Tasks.Feed/src/PushArtifactsInManifestToFeed.cs]: Haha yeah I know it's odd, I'll put something better together

File: src/Microsoft.DotNet.Build.Tasks.Feed/src/PushArtifactsInManifestToFeed.cs [hunk 3]
```diff
@@ -59,41 +98,35 @@ public class PushArtifactsInManifestToFeed : MSBuild.Task
...
+        /// The repo where we will file the issues when something fails while publishing.
+        /// arcade repo will be used by default.
         /// </summary>
-        [Required]
-        public string MaestroApiEndpoint { get; set; }
+        public string RepoForFilingPublishingIssues { get; set; } = "https://github.com/dotnet/arcade";
 
         /// <summary>
-        /// Authentication token to be used when interacting with Maestro API.
+        /// Who to tag in the created issue for them to investigate the failure.
```
Comment (riarenas) [inline src/Microsoft.DotNet.Build.Tasks.Feed/src/PushArtifactsInManifestToFeed.cs]:
```suggestion
        /// Who to tag in the created issue.
```

File: src/Microsoft.DotNet.Build.Tasks.Feed/Microsoft.DotNet.Build.Tasks.Feed.csproj [hunk 4]
```diff
@@ -3,7 +3,7 @@
 
   <PropertyGroup>
     <TargetFrameworks>netcoreapp2.1;net472</TargetFrameworks>
-
+    <SignAssembly>false</SignAssembly>
```
Comment (JohnTortugo) [inline src/Microsoft.DotNet.Build.Tasks.Feed/Microsoft.DotNet.Build.Tasks.Feed.csproj]: So we don't really need this? Wouldn't it cause some other project referencing it to fail?

**Comment by JohnTortugo** [comment]
@jcagme - just to see that I got it. This PR address only part of the issue, right? It doesn't include failure reporting for the case we fail to start the release pipeline, for instance.

**Comment by JohnTortugo** [comment]
Should we be concerned if the source repo isn't GitHub ?

**Comment by jcagme** [comment]
> @jcagme - just to see that I got it. This PR address only part of the issue, right? It doesn't include failure reporting for the case we fail to start the release pipeline, for instance.

True... and I don't want to copy/paste the code all over the place but given that this is a temporary workaround I don't know if creating a package is worth the effort. Let me think a bit about what's best

**Comment by jcagme** [comment]
> Should we be concerned if the source repo isn't GitHub ?

Yeah, I'll handle AzDO case as well

File: src/Microsoft.DotNet.Build.Tasks.Feed/Microsoft.DotNet.Build.Tasks.Feed.csproj (hunk 4 above)
Comment (jcagme) [reply src/Microsoft.DotNet.Build.Tasks.Feed/Microsoft.DotNet.Build.Tasks.Feed.csproj]: This just disabled SN signing. I confirmed with Tomas that it should not break anything. If an issue would exist then the validation repo (I think) will catch it

Comment (JohnTortugo) [reply src/Microsoft.DotNet.Build.Tasks.Feed/Microsoft.DotNet.Build.Tasks.Feed.csproj]: Thanks

**Comment by jcagme** [comment]
I've converted this change into it's own library which I've tested locally but I'm currently failing to test in an official build since the agents are not collaborating. Once I get pass through that and I'm able to validate I'll push the changes to this PR

File: Arcade.sln [hunk 5]
```diff
@@ -1,7 +1,7 @@
 ﻿
 Microsoft Visual Studio Solution File, Format Version 12.00
-# Visual Studio 15
-VisualStudioVersion = 15.0.28307.271
+# Visual Studio Version 16
```
Comment (JohnTortugo) [inline Arcade.sln]: Nit: was intentional?

File: src/Microsoft.DotNet.Git.IssueManager/src/IssueManager.cs [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+
+using Microsoft.DotNet.Git.IssueManager.Helpers;
+using System;
+using System.Threading.Tasks;
+
+namespace Microsoft.DotNet.Git.IssueManager
+{
+    public class IssueManager
+    {
+        public string GitHubPersonalAccessToken { get; set; }
```
Comment (JohnTortugo) [inline src/Microsoft.DotNet.Git.IssueManager/src/IssueManager.cs]: Will something access these properties directly?

File: src/Microsoft.DotNet.Git.IssueManager/src/IssueManager.cs [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.DotNet.Git.IssueManager.Helpers;
+using System;
+using System.Threading.Tasks;
+
+namespace Microsoft.DotNet.Git.IssueManager
+{
+    public class IssueManager
```
Comment (JohnTortugo) [inline src/Microsoft.DotNet.Git.IssueManager/src/IssueManager.cs]: Looks like this class just mostly forward method calls to RepositoryHelper. Can't we just do the calls directly to RepositoryHelper? I didn't look the whole implications of that.

File: src/Microsoft.DotNet.Git.IssueManager/src/Helpers/RepositoryHelper.cs [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.DotNet.Git.IssueManager.Clients;
+using System;
+using System.Threading.Tasks;
+
+namespace Microsoft.DotNet.Git.IssueManager.Helpers
+{
+    public class RepositoryHelper
```
Comment (JohnTortugo) [inline src/Microsoft.DotNet.Git.IssueManager/src/Helpers/RepositoryHelper.cs]: make the class static?

File: src/Microsoft.DotNet.Git.IssueManager/src/Clients/GitHubClient.cs [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+
+using Octokit;
+using System;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+
+namespace Microsoft.DotNet.Git.IssueManager.Clients
+{
+    public class GitHubClient
```
Comment (JohnTortugo) [inline src/Microsoft.DotNet.Git.IssueManager/src/Clients/GitHubClient.cs]: make class static?

File: src/Microsoft.DotNet.Git.IssueManager/src/Clients/AzureDevOpsClient.cs [hunk 10]
```diff
@@ -0,0 +1,88 @@
...
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+
+namespace Microsoft.DotNet.Git.IssueManager.Clients
+{
+    public class AzureDevOpsClient
```
Comment (JohnTortugo) [inline src/Microsoft.DotNet.Git.IssueManager/src/Clients/AzureDevOpsClient.cs]: static class?

File: src/Microsoft.DotNet.Build.Tasks.Feed/src/PushArtifactsInManifestToFeed.cs [hunk 11]
```diff
@@ -124,6 +126,9 @@ public async Task<bool> ExecuteAsync()
                 }
 
                 var buildModel = BuildManifestUtil.ManifestFileToModel(AssetManifestPath, Log);
+                commit = buildModel.Identity.Commit;
```
Comment (JohnTortugo) [inline src/Microsoft.DotNet.Build.Tasks.Feed/src/PushArtifactsInManifestToFeed.cs]: Is this still needed?

File: Arcade.sln (hunk 5 above)
Comment (jcagme) [reply Arcade.sln]: I do use VS16 but this change was not meant to be merged

File: src/Microsoft.DotNet.Git.IssueManager/src/IssueManager.cs (hunk 7 above)
Comment (jcagme) [reply src/Microsoft.DotNet.Git.IssueManager/src/IssueManager.cs]: This guy does not determine whether is a GH or AzDO call, that's done by the helper. All the helpers won't be visible to the user so she only will call IssueManager and the tool will manage to do the correct things

**Pull request merged**